src/utils/filterResults.ts

```diff
@@ -67,46 +83,59 @@ export default function filterResults (unsortedPropertyResults: UnprocessedResul
             const isUnderPropertyTypeLimit = matchTallies.propertyType < maxResultsForPropertyTypes
             const isUnderPropertyGroupLimit = matchTallies.propertyGroup < maxResultsForPropertyGroups
             const canAddBasedOnMaxResults = isUnderNeighbourLimit || isUnderPropertyTypeLimit || isUnderPropertyGroupLimit
-
             if (canAddBasedOnMaxResults) {
-                const propertyTypeMatch = !isPropertyTypeFilterInUse && isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)
-                const propertyGroupMatch = !isPropertyGroupFilterInUse && isUnderPropertyGroupLimit && matchForPropertyGroups(property, desiredPropertyGroups)
+                const propertyTypeMatch = isPropertyTypeFilterInUse && isUnderPropertyTypeLimit && matchForPropertyTypes(property, desiredPropertyTypes)
+                const propertyGroupMatch = isPropertyGroupFilterInUse && isUnderPropertyGroupLimit && matchForPropertyGroups(property, desiredPropertyGroups)
+
                 let canAddBasedOnFilters = propertyGroupMatch || propertyTypeMatch
+
                 if (filterInUse === 'SalesEvidenceFilter') {
+                    // N.B. when using sales evidence filter and type or group aren't used
+                    if (!isPropertyGroupFilterInUse && !isPropertyTypeFilterInUse) {
+                        canAddBasedOnFilters = true
+                    }
                     // N.B. the Sales Evidence Filter doesn't have the ability to search for multiple properties hence only passing in the single search param object.
-                    const allRecordsForSalesEvidenceFilter = searchParams.allRecords
                     canAddBasedOnFilters = allRecordsForSalesEvidenceFilter ? true : canAddBasedOnFilters && salesEvidenceFilter(searchParams, property)
                 }
 
-                const ownerData = getOwnerData(property)
                 const isManaged = (property.Managed === desiredManaged) || desiredManaged === 'All'
```

Comment:
I don't think all these if statements are needed
how about ```

const canAddBasedOnManaged = !isManagedFilterInUse || (property.Managed === desiredManaged) || desiredManaged === 'All'

const shouldAddProperty = canAddBasedOnManaged && canAddBasedOnFilters && isUnderNeighbourLimit
```

Reply:
I tried it and it doesn't work on the most basic test of simply selecting "Yes" for Managed and no other filter. So I'm not going to bother testing it for any of the other filter states.

If you want to find another way to get it to work differently to how it's currently working I'm happy for you to work on it. I've create a spreadsheet to show on what filter conditions the current setup passes. I can send you that so as to compare and remain consistent.

Reply:
Sorry should be ```const shouldAddProperty = canAddBasedOnManaged || canAddBasedOnFilters || isUnderNeighbourLimit```

share the spreadsheet when you get a chance. Would be good to reduce the cyclomatic complexity of this function to make it easier to understand and maintain

Reply:
I decided not to play around with it :P Too busy with other stuff today.